File: RepositoryPattern/app/src/main/java/com/example/vuclip/repositorypattern/dagger/AppComponent.java
package com.example.vuclip.repositorypattern.dagger;

import com.example.vuclip.repositorypattern.post.PostActivity;

import dagger.Component;

/**
 * Created by Banty on 08/04/18.
 */
@Component(modules = {MainActivityModule.class})
public interface AppComponent {
    void inject(PostActivity target);
}

File: DaggerRx/app/src/main/java/banty/com/daggerrx/di/component/AppComponent.kt
package banty.com.daggerrx.di.component

import banty.com.daggerrx.CryptoApp
import banty.com.daggerrx.di.module.AppModule
import banty.com.daggerrx.di.module.BuildersModule
import banty.com.daggerrx.di.module.NetworkModule
import dagger.Component
import dagger.android.AndroidInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AndroidInjectionModule::class,
        BuildersModule::class,
        AppModule::class,
        NetworkModule::class))
interface AppComponent {

    fun inject(app: CryptoApp)
}